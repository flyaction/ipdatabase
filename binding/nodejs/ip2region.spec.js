const path = require('path');
const IP2Region = require('./ip2region');

describe('ip2region', () => {
  let instance;

  beforeAll(() => {
    instance = IP2Region.create(path.join(__dirname, '../../data/ip2region.db'));
  });

  afterAll(() => {
    instance.destroy();
  });

  test('should query', () => {
    expect(instance.btreeSearchSync('120.24.78.68')).toMatchSnapshot();
    expect(instance.btreeSearchSync('10.10.10.10')).toMatchSnapshot();
  });

  test('binarySearch', () => {
    expect(instance.binarySearchSync('120.24.78.68')).toMatchSnapshot();
    expect(instance.binarySearchSync('10.10.10.10')).toMatchSnapshot();
  });
});
