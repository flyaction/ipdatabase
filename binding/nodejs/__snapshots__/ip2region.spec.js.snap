// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ip2region binarySearch 1`] = `
Object {
  "city": 2163,
  "region": "中国|0|广东省|深圳市|阿里云",
}
`;

exports[`ip2region binarySearch 2`] = `
Object {
  "city": 0,
  "region": "0|0|0|内网IP|内网IP",
}
`;

exports[`ip2region should query 1`] = `
Object {
  "city": 2163,
  "region": "中国|0|广东省|深圳市|阿里云",
}
`;

exports[`ip2region should query 2`] = `
Object {
  "city": 0,
  "region": "0|0|0|内网IP|内网IP",
}
`;
